src/provider/states/starting.rs

```diff
@@ -56,21 +56,37 @@ impl State<PodState> for Starting {
                     "Starting command: {:?} with arguments {:?}",
                     binary, os_args
                 );
+
+                // Check if environment variables were stated for the container
                 let env_variables = if let Some(vars) = container.env() {
                     debug!(
                         "Got environment vars: {:?} service {}",
                         vars, pod_state.service_name
                     );
-                    vars.iter()
+                    match vars
+                        .iter()
                         .map(|env_var| {
-                            (
-                                String::from(&env_var.name),
-                                String::from(
-                                    &env_var.value.clone().unwrap_or_else(|| String::from("")),
-                                ),
-                            )
+                            let name = String::from(&env_var.name);
+                            let value_result = CreatingConfig::render_config_template(
+                                &template_data,
+                                &env_var.value.clone().unwrap_or_else(|| String::from("")),
+                            );
+                            match value_result {
+                                Ok(value) => Ok((name, value)),
+                                Err(error) => Err(error),
+                            }
                         })
-                        .collect::<Vec<_>>()
+                        .collect()
+                    {
+                        // All variables were rendered correctly, continue
+                        Ok(rendered_value) => rendered_value,
+                        // One or more variables failed to render, transition this pod to a failed
+                        // state
+                        Err(error) => {
+                            error!("Failed to render value for env var due to: {:?}", error);
+                            return Transition::Complete(Err(anyhow::Error::from(error)));
+                        }
+                    }
```

Comment:
This bit looks suspicious as well.
Just passing on the value in the Ok(..) branch smells like `unwrap_or[_else]`

Reply:
On second thought this probably won't work because we need to return from an outer function from within an inner function...

Reply:
That was exactly the reason for doing this, returns from closures are not possible in Rust (well, they are possible, but don't return from the outer function (see https://internals.rust-lang.org/t/pre-rfc-allow-return-continue-and-break-to-affect-the-captured-environment/4791/34 )

After talking to the Ferrous systems people matches like this are fairly idiomatic, so I'll resolve this for now.